### DISCUSSION TITLE
Sending email when creating shipment via swell-node

### DISCUSSION THREAD

**Discussion by brahima** [swellstores/community#59] [Help] [answered]
Hello,

### Creating a shipment programmatically 
When I programmatically create a shipment to fulfill an order via my backend, no email is sent to warn the end user the order is shipped **unless** i specify a '$notify' parameter in my shipment payload. This is not documented but i found this parameter by inspecting what the swell dashboard was doing via Dev Tools. Without that option, no email is sent.

Here is my code, Note the "$notify" in the last line.

```
const ship = await this.swellClient.post('/shipments', {
      order_id: shipment.orderId,
      items: shipment.items.map(item =>
        ({ order_item_id: item.id, product_id: item.productId, quantity: 1 })
      ),
      tracking_code: shipment.trackingCode,
      service: shipment.serviceId,
      "$notify": "orders.shipped"
});
```

### Creating a shipment via the swell dashboard UI : 

<img width="1163" alt="image" src="https://user-images.githubusercontent.com/1575495/209666979-2a46c83a-87f5-46ce-93ab-05d478b485ad.png">

In the create/edit shipment (FULFILLMENT) popup from the order management (above screenshot), you can see that there is a '_Send email confirmation to customer_' checkbox. I understood from this checkbox, that sending an email was an option. This is checked by default when creating a shipment via the UI and unchecked by default when updating a shipment via the UI.

Hence, my question is :
- **Is it safe to use the $notify parameter the way i do ? If not, what's the alternative to have the order fulfilled mail sent to my customer when i create a shipment via the swell-node client ?**

*Note 1* : I was not able to find any [relevant documentation in the shipment section here](https://developers.swell.is/backend-api/shipments/create-a-shipment). There is this "notifications" key in the shipment model but it seems to be a read only value when getting shipments.

*Note 2* : I also noticed that when we're dealing with a shipment update, the $notify value was 'orders.shipped-update' and not 'orders.shipped' anymore (seems logical).

**Comment by greph** [comment, marked as answer]
Hi @brahima , you're exactly right: to send a notification when creating a shipment programmatically, you can include `"$notify": "orders.shipped"` in the request payload. 

I agree this would be great to have documented and have reached out to our content writing team to include in the docs. Thanks for your detailed comment!
